### ISSUE TITLE
Package manager seems broken

### ISSUE THREAD

**Issue by retsyo** [arturo-lang/arturo#388]
the closed issue https://github.com/arturo-lang/arturo/issues/359 says that the problem is fixed
but it is not true in the latest cloned and built arturo
 
```
$ /home/USER/.arturo/bin/arturo.exe -m list
- Launching Arturo module manager...

Error: Not enough arguments.

$ /home/USER/.arturo/bin/arturo.exe -m install html
- Launching Arturo module manager...

Error: Not enough arguments.
```

**Comment by drkameleon** [comment]
Hmm... Good catch. I thought it was 😞 

I guess I'll have to look into. Thanks for pointing it out! 👍

**Comment by xerox51** [comment]
I have meet same problem on windows using arturo full version.
But I use brew install arturo instruction to install arturo on mac, it works normally.

**Comment by drkameleon** [comment]
> I have meet same problem on windows using arturo full version. But I use brew install arturo instruction to install arturo on mac, it works normally.

Hmm... Very interesting. So, it appears to do with Mini vs Full version. Weird.

I'll look into it. Thanks a lot for letting me know! 😄

**Comment by xerox51** [comment]
> > I have meet same problem on windows using arturo full version. But I use brew install arturo instruction to install arturo on mac, it works normally.
> 
> Hmm... Very interesting. So, it appears to do with Mini vs Full version. Weird.
> 
> I'll look into it. Thanks a lot for letting me know! 😄

It's not the question about full or min. I just tried to report to you using pre-bulit binary arturo 0.9.78 full version on windows will encouter the problem when you type the arturo -m list command line in ihe cmd window.
And after I used brew install arturo command to install arturo on mac, I didn't encouter the problem which I meet on windows.

**Comment by retsyo** [comment]
if I run
```
/home/USER/.arturo/bin/arturo.exe  src/scripts/module.art list
```
I get 
```
- Launching Arturo module manager...
- Fetching list...

> Available local modules:
   ! No modules found
```

so, I think, the code to call module.art in arturo does not pass arguments correctly. Then I add
```
        echo "arguments = ", arguments
```
after 
```
        arguments = token.remainingArgs()
```
the commnad
```
$ /home/USER/.arturo/bin/arturo.exe -m list
```
says
```
arguments = @[]
- Launching Arturo module manager...

Error: Not enough arguments.
```
so it must be something wrong in code dealing with `token`.

**Comment by drkameleon** [comment]
I think I have totally spotted the issue: basically, it's what you say. The argument passing and parsing in Windows doesn't work exactly the same way it's supposed to for Mac/Linux.

Let's see. I'll work on that! :)

**Comment by skydive241** [comment]
Do not consume remaining arguments in arturo.nim - correction like

```
...
of "m", "module":
    action = evalCode
    code = runModule
    break                                  # <==
...
```
solves this issue.

Greetings from germany
Patrick

**Comment by drkameleon** [comment]
> Do not consume remaining arguments in arturo.nim - correction like
> 
> ```
> ...
> of "m", "module":
>     action = evalCode
>     code = runModule
>     break                                  # <==
> ...
> ```
> 
> solves this issue.
> 
> Greetings from germany Patrick

Patrick, Hi!

I haven't managed to find the time to work on it, but - without even trying it out - I believe you are spot-on! 😉 

Thanks for letting me know.

I'll have a look into it and try to fix it asap.

Cheers from Granada, Spain! 😄

**Comment by drkameleon** [comment]
Just closed it (automatically) as I think @skydive241 's  suggestion (thanks! :wink) worked beautifully.

I also tried verifying it via the CI build and it seems to be working fine (on Windows):
<img width="818" alt="Screenshot 2022-02-07 at 16 32 20" src="https://user-images.githubusercontent.com/1265028/152819282-3daac3d6-5a04-48df-b349-7b84e4fe60b2.png">

The package manager definitely leaves a lot to be desired - basically, it asks for some complete rewrite and re-thinking of how it should be. That being said, for now the issue in question appears to have been solved. (If not, just let me know and I'll re-open it 😄 )
